superset-frontend/spec/javascripts/middleware/asyncEvent_spec.ts

```diff
@@ -17,249 +17,203 @@
  * under the License.
  */
 import fetchMock from 'fetch-mock';
+import WS from 'jest-websocket-mock';
 import sinon from 'sinon';
 import * as featureFlags from 'src/featureFlags';
-import initAsyncEvents from 'src/middleware/asyncEvent';
-
-jest.useFakeTimers();
+import { parseErrorJson } from 'src/utils/getClientErrorObject';
+import * as asyncEvent from 'src/middleware/asyncEvent';
 
 describe('asyncEvent middleware', () => {
-  const next = sinon.spy();
-  const state = {
-    charts: {
-      123: {
-        id: 123,
-        status: 'loading',
-        asyncJobId: 'foo123',
-      },
-      345: {
-        id: 345,
-        status: 'loading',
-        asyncJobId: 'foo345',
-      },
-    },
+  const asyncPendingEvent = {
+    status: 'pending',
+    result_url: null,
+    job_id: 'foo123',
+    channel_id: '999',
+    errors: [],
+  };
+  const asyncDoneEvent = {
+    id: '1518951480106-0',
+    status: 'done',
+    result_url: '/api/v1/chart/data/cache-key-1',
+    job_id: 'foo123',
+    channel_id: '999',
+    errors: [],
   };
-  const events = [
-    {
-      status: 'done',
-      result_url: '/api/v1/chart/data/cache-key-1',
-      job_id: 'foo123',
-      channel_id: '999',
-      errors: [],
-    },
-    {
-      status: 'done',
-      result_url: '/api/v1/chart/data/cache-key-2',
-      job_id: 'foo345',
-      channel_id: '999',
-      errors: [],
-    },
-  ];
-  const mockStore = {
-    getState: () => state,
-    dispatch: sinon.stub(),
+  const asyncErrorEvent = {
+    id: '1518951480107-0',
+    status: 'error',
+    result_url: null,
+    job_id: 'foo123',
+    channel_id: '999',
+    errors: [{ message: "Error: relation 'foo' does not exist" }],
   };
-  const action = {
-    type: 'GENERIC_ACTION',
+  const chartData = {
+    some: 'data',
```

Comment:
nit: This comes down to personal preference, but for me it's always easier to digest tests/mocks if they resemble real requests/responses as much as possible. In this case I believe `chartData` should be of type `ChartDataResponseResult[]`: https://github.com/apache-superset/superset-ui/blob/a3ec94f7a72a3dbb6cb6563e7ea7d3582661ea43/packages/superset-ui-core/src/query/types/QueryResponse.ts#L46-L72